src/test/java/org/junit/platform/engine/test/event/ExecutionEventRecorder.java

```diff
@@ -0,0 +1,175 @@
+		addEvent(ExecutionEvent.executionStarted(testDescriptor));
+	}
+
+	@Override
+	public void executionFinished(TestDescriptor testDescriptor, TestExecutionResult result) {
+		addEvent(ExecutionEvent.executionFinished(testDescriptor, result));
+	}
+
+	@Override
+	public void reportingEntryPublished(TestDescriptor testDescriptor, ReportEntry entry) {
+		addEvent(ExecutionEvent.reportingEntryPublished(testDescriptor, entry));
+	}
+
+	public List<ExecutionEvent> getExecutionEvents() {
+		return executionEvents;
+	}
+
+	public Stream<ExecutionEvent> eventStream() {
+		return getExecutionEvents().stream();
+	}
+
+	public long getTestSkippedCount() {
+		return testEventsByType(SKIPPED).count();
+	}
+
+	public long getTestStartedCount() {
+		return testEventsByType(STARTED).count();
+	}
+
+	public long getReportingEntryPublishedCount() {
+		return testEventsByType(REPORTING_ENTRY_PUBLISHED).count();
+	}
+
+	public long getDynamicTestRegisteredCount() {
+		return testEventsByType(DYNAMIC_TEST_REGISTERED).count();
+	}
+
+	public long getTestFinishedCount() {
+		return testEventsByType(FINISHED).count();
+	}
+
+	public long getTestSuccessfulCount() {
+		return getTestFinishedCount(Status.SUCCESSFUL);
+	}
+
+	public long getTestAbortedCount() {
+		return getTestFinishedCount(Status.ABORTED);
+	}
+
+	public long getTestFailedCount() {
+		return getTestFinishedCount(Status.FAILED);
+	}
+
+	public long getContainerSkippedCount() {
+		return containerEventsByType(SKIPPED).count();
+	}
+
+	public long getContainerStartedCount() {
+		return containerEventsByType(STARTED).count();
+	}
+
+	public long getContainerFinishedCount() {
+		return containerEventsByType(FINISHED).count();
+	}
+
+	public List<ExecutionEvent> getSkippedTestEvents() {
+		return testEventsByType(Type.SKIPPED).collect(toList());
+	}
+
+	public List<ExecutionEvent> getSuccessfulTestFinishedEvents() {
+		return testFinishedEvents(Status.SUCCESSFUL).collect(toList());
+	}
+
+	public List<ExecutionEvent> getFailedTestFinishedEvents() {
+		return testFinishedEvents(Status.FAILED).collect(toList());
+	}
+
+	private long getTestFinishedCount(Status status) {
+		return testFinishedEvents(status).count();
+	}
+
+	private Stream<ExecutionEvent> testFinishedEvents(Status status) {
+		return testEventsByType(FINISHED).filter(
+			byPayload(TestExecutionResult.class, where(TestExecutionResult::getStatus, isEqual(status))));
+	}
+
+	private Stream<ExecutionEvent> testEventsByType(Type type) {
+		return eventsByTypeAndTestDescriptor(type, TestDescriptor::isTest);
+	}
+
+	private Stream<ExecutionEvent> containerEventsByType(Type type) {
+		return eventsByTypeAndTestDescriptor(type, TestDescriptor::isContainer);
+	}
+
+	private Stream<ExecutionEvent> eventsByTypeAndTestDescriptor(Type type,
+			Predicate<? super TestDescriptor> predicate) {
+		return eventStream().filter(byType(type).and(byTestDescriptor(predicate)));
+	}
+
+	private void addEvent(ExecutionEvent event) {
+		executionEvents.add(event);
+	}
+
+}
```

Comment:
I think it would be worth making the constructor `private` and allowing instances to be created with `ExecutionEventRecorder.create()`.